src/globus_sdk/transport/retry.py

```diff
@@ -0,0 +1,254 @@
+        - default_check_request_exception
+        - default_check_retry_after_header
+        - default_check_tranisent_error
+        - default_check_expired_authorization
+
+        It can be overridden to register additional hooks or to remove the default
+        hooks.
+        """
+        self.register_check(self.default_check_max_retries_exceeded)
+        self.register_check(self.default_check_request_exception)
+        self.register_check(self.default_check_retry_after_header)
+        self.register_check(self.default_check_transient_error)
+        self.register_check(self.default_check_expired_authorization)
+
+    def default_check_max_retries_exceeded(
+        self, ctx: RetryContext
+    ) -> typing.Optional[bool]:
+        """check if the max retries for this policy have been exceeded"""
+        return False if ctx.attempt >= self.max_retries else None
+
+    def default_check_request_exception(
+        self, ctx: RetryContext
+    ) -> typing.Optional[bool]:
+        """check if a network error was encountered"""
+        if not ctx.exception:
+            return None
+        return isinstance(ctx.exception, requests.RequestException)
+
+    def default_check_retry_after_header(
+        self, ctx: RetryContext
+    ) -> typing.Optional[bool]:
+        """check for a retry-after header if the response had a matching status"""
+        if (
+            ctx.response is None
+            or ctx.response.status_code not in self.RETRY_AFTER_STATUS_CODES
+        ):
+            return None
+        retry_after = _parse_retry_after(ctx.response)
+        if retry_after:
+            ctx.backoff = float(retry_after)
+        return True
+
+    def default_check_transient_error(self, ctx: RetryContext) -> typing.Optional[bool]:
+        """check for transient error status codes which could be resolved by retrying
+        the request"""
+        if (
+            ctx.response is None
+            or ctx.response.status_code not in self.TRANSIENT_ERROR_STATUS_CODES
+        ):
+            return None
+        return True
+
+    def default_check_expired_authorization(
+        self, ctx: RetryContext
+    ) -> typing.Optional[bool]:
+        """check for expired authorization, as represented by a 401 error when the
+        authorizer supports handling for missing/invalid authorization"""
+        if ctx.response is None:
+            return None
+        if ctx.response.status_code not in self.EXPIRED_AUTHORIZATION_STATUS_CODES:
+            return None
+        if ctx.authorizer is None:
+            return None
+        # the response code was a 401 and there's already been at least one reauth
+        # attempt
+        # this could be `False` instead, but that would mean that nobody else can add
+        # 401-retries even when reauth didn't work
+        if ctx.retry_state.get("has_done_reauth"):
+            return None
+
+        if not ctx.authorizer.handle_missing_authorization():
+            return None
```

Comment:
Comments aside, could these all be lumped into one large conditional?

Reply:
They could, but I worry that combining _all_ of them would be optimizing for lines-of-code rather than clarity. 

Because `handle_missing_authorization` can have side-effects, I think it's better to keep that one separate.

How about this?

```python
if (  # is this check applicable?
    ctx.response is None or
    ctx.response.status_code not in self.EXPIRED_AUTHORIZATION_STATUS_CODES or
    ctx.authorizer is None
):
    return None

# ... blah blah long comment ...
if ctx.retry_state.get("has_done_reauth"):
    return None

# potentially do a token refresh, exiting if the authorizer can't or won't handle this
if not ctx.authorizer.handle_missing_authorization():
    return None
```

Reply:
Actually, I went ahead and did this refactor since I already needed to touch this to add the result enum.